wfe/wfe.go

```diff
@@ -858,6 +858,9 @@ func (wfe *WebFrontEndImpl) verifyPOST(
 	body := string(bodyBytes)
 	parsedJWS, err := wfe.parseJWS(body)
 	if err != nil {
+		if strings.Contains(err.Error(), "unexpected signature algorithm") {
```

Comment:
Rather than checking the contents of the error string, I think it would be better for `wfe.parseJWS` to behave the same as `wfe.verifyJWS` and simply return a ProblemDetails directly instead of returning an error. Then parseJWS can decide which acme error is most appropriate in each case.

(Note, this is the opposite of how Boulder does things. But pebble doesn't have boulder's automatic err-to-problem conversion.)

Reply:
is it safe to update requried go version to 1.24 to use new go-jose? othersiwe string match on error is required as that is only way to catch signiture algorithm error from current version. this error is sent from go-jose, not parseJWS

Reply:
updated go version to used with newer go-jose: if one revert to 6b7368b this can be cancel